### PR TITLE
Add the ability to tag passed boxes

### PR DESCRIPTION
**Pull request by rene-ph**
The final result is here! #2 
ExampleTags

New things:

1) There is a new component called tag where it resides all the logic needed to display the tags and the input 
2) The new file called chromeStorage.js was created  to be able to manipulate in one place all the "transactions" like get the tags, remove the tags and update the tags. Also there are some generic methods that can be reused as well. 
3) The way is storing right now the tags  it's the same way you mention about making an array of arrays. 
Example:
```
[
  {
    "end": "2020-09-10T23:25:46.446Z",
    "id": 1602631571951,
    "name": "Goal",
    "start": "2020-08-30T23:25:46.446Z",
    "tag": [
      [],
      [
        "Tag1",
        "Tag2"
      ],
      [
        "AnewTag"
      ],
      [],
      [],
      [],
      [],
      [],
      [],
      [],
      []
    ]
  }
]

```

Diagrama en blanco

Let me know your thoughts! 
Thanks
Regards!

### CODE REVIEW

File: src/db/chromeSotrage.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*global chrome*/
+export const ChromeStorage =  {
+     getAllData: () => {
+        return new Promise((resolve, reject) => {
+                chrome.storage.sync.get(null, (data)=>{ resolve(data)});
+        });
+    },
+
+    saveData: (key, value) => {
+        chrome.storage.sync.set({[`${key}`]: value}, ()=>{});
```
Comment (aladhims) [inline src/db/chromeSotrage.js:10]: I think this needs to provide 3rd param which is the callback function, so the caller can have custom callback function for its own usecase. 

Probably something like this?
```javascript
saveData: (key, value, callback) => {
        chrome.storage.sync.set({[`${key}`]: value}, callback);
}
````

File: src/db/chromeSotrage.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    },
+
+    removeData: (key) => {
+        chrome.storage.sync.set({[`${key}`]: null}, ()=>{});
```
Comment (aladhims) [inline src/db/chromeSotrage.js:14]: same as above

File: src/visualizer/components/tag.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  },[refInput]);
+  
+
+  const handleInputChange = e => {
+    setInputValue(e.target.value);
+  };
+
+
+  const updateSotrageTag = (tag) => {;
```
Comment (aladhims) [inline src/visualizer/components/tag.js:50]: a typo here

**Review by aladhims — CHANGES_REQUESTED** [review]
@rene-ph  Thanks for making a complete, descriptive, clean PR. This PR is inevitably perfect! Although I have some comments above which probably needs some little adjustments, for the most part, I'm already amazed.

File: src/db/chromeSotrage.js (hunk 1 above)
Comment (rene-ph) [reply src/db/chromeSotrage.js]: Fixed!

File: src/visualizer/components/tag.js (hunk 3 above)
Comment (rene-ph) [reply src/visualizer/components/tag.js]: Fixed

**Pull request merged**